localization/util/pose_initializer/src/pose_initializer_core.cpp

```diff
@@ -38,47 +41,55 @@ double getGroundHeight(const pcl::PointCloud<pcl::PointXYZ>::Ptr pcdmap, const t
   return std::isfinite(height) ? height : point.getZ();
 }
 
-PoseInitializer::PoseInitializer(ros::NodeHandle nh, ros::NodeHandle private_nh)
-: nh_(nh), private_nh_(private_nh), tf2_listener_(tf2_buffer_), map_frame_("map")
+PoseInitializer::PoseInitializer()
+: Node("pose_initializer"), tf2_listener_(tf2_buffer_), map_frame_("map")
 {
-  initial_pose_sub_ = nh_.subscribe("initialpose", 10, &PoseInitializer::callbackInitialPose, this);
-  map_points_sub_ = nh_.subscribe("pointcloud_map", 1, &PoseInitializer::callbackMapPoints, this);
-
-  bool use_first_gnss_topic = true;
-  private_nh_.getParam("use_first_gnss_topic", use_first_gnss_topic);
+  // We can't use _1 because pcl leaks an alias to boost::placeholders::_1, so it would be ambiguous
+  initial_pose_sub_ = this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+    "initialpose", 10,
+    std::bind(&PoseInitializer::callbackInitialPose, this, std::placeholders::_1));
+  map_points_sub_ = this->create_subscription<geometry_msgs::msg::PoseWithCovarianceStamped>(
+    "pointcloud_map", 1,
+    std::bind(&PoseInitializer::callbackMapPoints, this, std::placeholders::_1));
+
+  const bool use_first_gnss_topic = this->declare_parameter("use_first_gnss_topic", true);
   if (use_first_gnss_topic) {
-    gnss_pose_sub_ = nh_.subscribe("gnss_pose_cov", 1, &PoseInitializer::callbackGNSSPoseCov, this);
+    gnss_pose_sub_ = this->create_subscription<sensor_msgs::msg::PointCloud2>(
+      "gnss_pose_cov", 1,
+      std::bind(&PoseInitializer::callbackGNSSPoseCov, this, std::placeholders::_1));
   }
 
-  initial_pose_pub_ = nh_.advertise<geometry_msgs::PoseWithCovarianceStamped>("initialpose3d", 10);
+  initial_pose_pub_ =
+    this->create_publisher<geometry_msgs::msg::PoseWithCovarianceStamped>("initialpose3d", 10);
 
-  ndt_client_ =
-    nh_.serviceClient<autoware_localization_srvs::PoseWithCovarianceStamped>("ndt_align_srv");
-  ndt_client_.waitForExistence(ros::Duration(1.0));  // TODO
+  ndt_client_ = this->create_client<autoware_localization_srvs::srv::PoseWithCovarianceStamped>(
+    "ndt_align_srv");
+  ndt_client_->wait_for_service(std::chrono::seconds(1));  // TODO
```

Comment:
Not sure if the TODO refers to that, but this function returns a bool that could be used to check if the service went up.

Reply:
Yes, I think this TODO refers to error check.
Since we have no error handling when we call the service anymore, we might want to change this so that it waits until the service is actually available.

e.g.
```cpp
  while (!ndt_client_->wait_for_service(std::chrono::seconds(1)) && rclcpp::ok()) {
    RCLCPP_INFO(get_logger(), "Waiting for service...");
  }
```

Reply:
:+1: Done.